test/functional/services/dashboard/expectations.ts

```diff
@@ -32,11 +32,32 @@ export class DashboardExpectService extends FtrService {
   }
 
   async visualizationsArePresent(vizList: string[]) {
-    this.log.debug('Checking all visualisations are present on dashsboard');
+    this.log.debug('Checking all visualisations are present on the dashboard');
     const notLoaded = await this.dashboard.getNotLoadedVisualizations(vizList);
     expect(notLoaded).to.be.empty();
   }
 
+  /**
+   * Asserts that there is no error embeddables on the dashboard
+   * @throws An error if an error embeddable is present
+   */
+  async noErrorEmbeddablesPresent() {
+    this.log.debug('Ensure that there are no error embeddables on the dashboard');
+    const errorEmbeddables = await this.testSubjects.findAll('embeddableError');
+    if (errorEmbeddables.length > 0) {
+      const errorTitles = await Promise.all(
+        errorEmbeddables.map(async (embeddable) => {
+          return (
+            (await (await embeddable.findByTestSubject('dashboardPanelTitle')).getVisibleText()) ??
+            'Empty title'
+          );
```

Comment:
I thought about taking a screenshot of **every** error embeddable like so:
```suggestion
          const panelTitle =
            (await (await embeddable.findByTestSubject('dashboardPanelTitle')).getVisibleText()) ??
            'Empty title';
          await embeddable.scrollIntoViewIfNecessary();
          await this.screenshotService.takeForFailure(panelTitle);
          return panelTitle;
```

But I'm not sure if that is overkill? If every single panel fails (such as when the testing environment doesn't load properly), generating such a large number of failure screenshots could very quickly get out of hand 🤷